src/overlay/OverlayManagerImpl.cpp

```diff
@@ -1300,15 +1278,13 @@ OverlayManagerImpl::demand()
     }
 
     auto authenticatedPeers = getAuthenticatedPeers();
-    std::vector<Peer::pointer> pullModePeers;
-    for (auto const& peer : authenticatedPeers)
-    {
-        if (peer.second->isPullModeEnabled())
-        {
-            pullModePeers.push_back(peer.second);
-        }
-    }
-    stellar::shuffle(pullModePeers.begin(), pullModePeers.end(), gRandomEngine);
+    std::vector<Peer::pointer> peers;
```

Comment:
We should use `OverlayManagerImpl::getRandomAuthenticatedPeers` here instead

Reply:
I didn't know about this! Thanks!